api/clients/v2/accountant.go

```diff
@@ -74,50 +73,76 @@ func (a *Accountant) BlobPaymentInfo(
 	quorumNumbers []uint8,
 	timestamp int64) (*big.Int, error) {
 
+	symbolUsage := meterer.SymbolsCharged(numSymbols, a.minNumSymbols)
+
+	// Always try to use reservation first
+	payment, err := a.ReservationUsage(symbolUsage, quorumNumbers, timestamp)
+	if err == nil {
+		return payment, nil
+	}
+
+	// Fall back to on-demand payment if reservation fails
+	return a.OnDemandUsage(symbolUsage, quorumNumbers)
+}
+
+// ReservationUsage attempts to use the reservation for the given request.
+// Returns (0, nil) if successful, or (nil, error) if reservation cannot be used.
+func (a *Accountant) ReservationUsage(
+	symbolUsage uint64,
+	quorumNumbers []uint8,
+	timestamp int64) (*big.Int, error) {
+
 	currentReservationPeriod := meterer.GetReservationPeriodByNanosecond(timestamp, a.reservationWindow)
-	symbolUsage := a.SymbolsCharged(numSymbols)
 
 	a.usageLock.Lock()
 	defer a.usageLock.Unlock()
+
 	relativePeriodRecord := a.GetRelativePeriodRecord(currentReservationPeriod)
 	relativePeriodRecord.Usage += symbolUsage
 
-	// first attempt to use the active reservation
+	// Check if we can use the reservation within the bin limit
 	binLimit := a.reservation.SymbolsPerSecond * uint64(a.reservationWindow)
 	if relativePeriodRecord.Usage <= binLimit {
-		if err := QuorumCheck(quorumNumbers, a.reservation.QuorumNumbers); err != nil {
-			return big.NewInt(0), err
+		if err := meterer.ValidateQuorum(quorumNumbers, a.reservation.QuorumNumbers); err != nil {
+			return nil, err
 		}
 		return big.NewInt(0), nil
 	}
 
+	// Try to use overflow bin if available
 	overflowPeriodRecord := a.GetRelativePeriodRecord(currentReservationPeriod + 2)
-	// Allow one overflow when the overflow bin is empty, the current usage and new length are both less than the limit
 	if overflowPeriodRecord.Usage == 0 && relativePeriodRecord.Usage-symbolUsage < binLimit && symbolUsage <= binLimit {
-		if err := QuorumCheck(quorumNumbers, a.reservation.QuorumNumbers); err != nil {
-			return big.NewInt(0), err
+		if err := meterer.ValidateQuorum(quorumNumbers, a.reservation.QuorumNumbers); err != nil {
+			return nil, err
 		}
 		overflowPeriodRecord.Usage += relativePeriodRecord.Usage - binLimit
 		return big.NewInt(0), nil
 	}
 
-	// reservation not available, rollback reservation records, attempt on-demand
-	//todo: rollback on-demand if disperser respond with some type of rejection?
```

Comment:
Is this todo captured somewhere / resolved?

Reply:
added to [commit](https://github.com/Layr-Labs/eigenda/pull/1653/commits/510fe1a875d031d2a9ce022ad8751582719e0f85); probably not going to be done anytime soon